CRM/Event/Form/Registration/Confirm.php

```diff
@@ -454,11 +454,10 @@ public function postProcess() {
       }
       if ($this->isShowPaymentOnConfirm()) {
         // "is_pay_later" may have been set by the registration page. Reset it here.
-        $params[$participantNum]['is_pay_later'] = 0;
-        // Again, here we have to use getSubmitValue because getSubmittedValue is not set.
-        if ($this->getSubmitValue('hidden_processor') === NULL || $this->getSubmitValue('payment_processor_id') == 0) {
+        $params[$participantNum]['is_pay_later'] = $this->_values['event']['is_pay_later'] = 0;
+        if ($this->getSubmittedValue('payment_processor_id') == 0) {
           // If we submitted with no payment processor then we must be pay later - set it here.
-          $params[$participantNum]['is_pay_later'] = 1;
+          $params[$participantNum]['is_pay_later'] = $this->_values['event']['is_pay_later'] = 1;
```

Comment:
Set the pay later status based on the payment processor used on the confirm form